File: backend/app/schemas/product.py
from decimal import Decimal

from pydantic import BaseModel


class PriceHistoryPoint(BaseModel):
    price: Decimal
    mrp: Decimal | None = None
    in_stock: bool
    captured_at: str


class PriceHistoryResponse(BaseModel):
    platform: str
    product_id: str
    days: int
    history: list[PriceHistoryPoint]

File: mobile/src/types/product.ts
export interface PlatformPrice {
  platform: string;
  price: number;
  mrp: number | null;
  discount_pct: number | null;
  in_stock: boolean;
  delivery_eta_minutes: number | null;
  platform_product_id: string;
}

export interface ComparedProduct {
  name: string;
  brand: string | null;
  unit_quantity: string | null;
  image_url: string | null;
  prices: PlatformPrice[];
  best_price: PlatformPrice | null;
}

export interface PlatformStatus {
  status: string;
  response_time_ms: number | null;
  message: string | null;
}

export interface SearchResponse {
  query: string;
  results: ComparedProduct[];
  platform_status: Record<string, PlatformStatus>;
}

export interface PriceHistoryPoint {
  price: number;
  mrp: number | null;
  in_stock: boolean;
  captured_at: string;
}
